src/utils/handleOptions.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Evaluates the value of a moldObject property.
+ * If the value is a function, it is called with the filtered options object as the its parameter.
+ * Otherwise, the value is returned as is.
+ * @param {object} filteredObject - the current filtered options object
+ * @param {any} value - the value to evaluate
+ * @returns {any} the evaluated value
+ */
+function evaluate(filteredObject, value) {
+  if (typeof value === "function") {
+    return value(filteredObject);
+  } else {
+    return value;
+  }
+}
+
+/**
+ * Checks if a value is within the enumArray.
+ * If isCaseInsensitive is true, the comparison is case insensitive.
+ *
+ * @param {any} value - the value to check
+ * @param {Array} enumArray - the array of enum values
+ * @param {boolean} isCaseInsensitive - whether the comparison is case insensitive
+ * @returns {any | undefined} the enum value if the user value is within the enumArray, otherwise undefined
+ */
+function checkEnum(value, enumArray, isCaseInsensitive) {
+  if (isCaseInsensitive && typeof value === "string") {
+    for (const enumValue of enumArray) {
+      if (enumValue.toLowerCase() === userValue.toLowerCase()) {
+        return enumValue;
+      }
+    }
+  } else {
+    for (const enumValue of enumArray) {
+      if (enumValue === value) {
+        return enumValue;
+      }
+    }
+  }
```

Comment:
I would be inclined to write the loop once and have the `if` condition inside rather than outside.

Alternatively, there are modern JS array methods which can be used to simplify the logic like `.some()`, `.includes()`, and `.find()`.  For example, 
```
// check for exact match
const isValid = enumArray.includes(value);
```
```
// check for case-insensitive match
const matchingVal = enumArray.find(enumValue => enumValue.toLowerCase() === value.toLowerCase());
```

Looks like `userValue` is a typo and should be `value`?

I wonder if enum options in the codebase are always `string` or if there are some `number` enums?